### PR TITLE
New template ReactNativeTypeScriptTemplate

### PR DESCRIPTION
**Pull request by wmathurin**
Basic react native application written in typescript

### CODE REVIEW

File: ReactNativeTypeScriptTemplate/app.tsx [hunk 1]
```diff
@@ -0,0 +1,117 @@
+/*
```
Comment (wmathurin) [inline ReactNativeTypeScriptTemplate/app.tsx:1]: Compare to https://github.com/forcedotcom/SalesforceMobileSDK-Templates/blob/dev/ReactNativeTemplate/app.js

File: ReactNativeTypeScriptTemplate/.buckconfig [hunk 2]
```diff
@@ -0,0 +1,6 @@
+
+[android]
+  target = Google Inc.:Google APIs:27
```
Comment (bhariharan) [inline ReactNativeTypeScriptTemplate/.buckconfig:3]:
```suggestion
  target = Google Inc.:Google APIs:30
```

File: ReactNativeTypeScriptTemplate/android/app/src/main/java/com/salesforce/reactnativetemplate/MainActivity.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2015-present, salesforce.com, inc.
```
Comment (bhariharan) [inline ReactNativeTypeScriptTemplate/android/app/src/main/java/com/salesforce/reactnativetemplate/MainActivity.java:2]:
```suggestion
 * Copyright (c) 2020-present, salesforce.com, inc.
```

File: ReactNativeTypeScriptTemplate/android/app/src/main/java/com/salesforce/reactnativetemplate/MainApplication.java [hunk 4]
```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2015-present, salesforce.com, inc.
```
Comment (bhariharan) [inline ReactNativeTypeScriptTemplate/android/app/src/main/java/com/salesforce/reactnativetemplate/MainApplication.java:2]:
```suggestion
 * Copyright (c) 2020-present, salesforce.com, inc.
```

File: ReactNativeTypeScriptTemplate/android/app/src/main/java/com/salesforce/reactnativetemplate/MainApplication.java [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+         * Uncomment the following line to enable IDP login flow. This will allow the user to
+         * either authenticate using the current app or use a designated IDP app for login.
+         * Replace 'idpAppURIScheme' with the URI scheme of the IDP app meant to be used.
+         */
+		// SalesforceReactSDKManager.getInstance().setIDPAppURIScheme(idpAppURIScheme);
+
+        /*
+		 * Un-comment the line below to enable push notifications in this app.
+		 * Replace 'pnInterface' with your implementation of 'PushNotificationInterface'.
+		 * Add your Google package ID in 'bootonfig.xml', as the value
```
Comment (bhariharan) [inline ReactNativeTypeScriptTemplate/android/app/src/main/java/com/salesforce/reactnativetemplate/MainApplication.java:90]:
```suggestion
		 * Add your Google package ID in 'bootconfig.xml', as the value
```

File: ReactNativeTypeScriptTemplate/android/keystores/debug.keystore.properties [hunk 6]
```diff
@@ -0,0 +1,4 @@
+key.store=debug.keystore
```
Comment (bhariharan) [inline ReactNativeTypeScriptTemplate/android/keystores/debug.keystore.properties:1]: Shouldn't the `keystores/` folder be added to `.gitignore`?

File: ReactNativeTypeScriptTemplate/app.tsx [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+                    (error) => console.log('Failed to authenticate:' + error)
+                );
+            });
+    }
+
+    fetchData() {
+        var that = this;
+        net.query('SELECT Id, Name FROM Contact LIMIT 100',
+                  (response:Response) => that.setState({data: response.records}),
+                  (error) => console.log('Failed to querry:' + error)
```
Comment (bhariharan) [inline ReactNativeTypeScriptTemplate/app.tsx:77]:
```suggestion
                  (error) => console.log('Failed to query:' + error)
```

File: ReactNativeTypeScriptTemplate/index.js [hunk 8]
```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 2017-present, salesforce.com, inc.
```
Comment (bhariharan) [inline ReactNativeTypeScriptTemplate/index.js:2]:
```suggestion
 * Copyright (c) 2020-present, salesforce.com, inc.
```

File: ReactNativeTypeScriptTemplate/ios/ReactNativeTypeScriptTemplate.xcodeproj/project.pbxproj [hunk 9]
```diff
@@ -0,0 +1,460 @@
...
+			productType = "com.apple.product-type.application";
+		};
+/* End PBXNativeTarget section */
+
+/* Begin PBXProject section */
+		83CBB9F71A601CBA00E9B192 /* Project object */ = {
+			isa = PBXProject;
+			attributes = {
+				LastUpgradeCheck = 1020;
+				ORGANIZATIONNAME = Facebook;
```
Comment (bhariharan) [inline ReactNativeTypeScriptTemplate/ios/ReactNativeTypeScriptTemplate.xcodeproj/project.pbxproj:150]:
```suggestion
				ORGANIZATIONNAME = Salesforce.com;
```

File: ReactNativeTypeScriptTemplate/ios/ReactNativeTypeScriptTemplate/AppDelegate.h [hunk 10]
```diff
@@ -0,0 +1,32 @@
+/*
+ Copyright (c) 2015-present, salesforce.com, inc. All rights reserved.
```
Comment (bhariharan) [inline ReactNativeTypeScriptTemplate/ios/ReactNativeTypeScriptTemplate/AppDelegate.h:2]:
```suggestion
 Copyright (c) 2020-present, salesforce.com, inc. All rights reserved.
```

File: ReactNativeTypeScriptTemplate/ios/ReactNativeTypeScriptTemplate/AppDelegate.m [hunk 11]
```diff
@@ -0,0 +1,175 @@
+/*
+ Copyright (c) 2011-present, salesforce.com, inc. All rights reserved.
```
Comment (bhariharan) [inline ReactNativeTypeScriptTemplate/ios/ReactNativeTypeScriptTemplate/AppDelegate.m:2]:
```suggestion
 Copyright (c) 2020-present, salesforce.com, inc. All rights reserved.
```

File: ReactNativeTypeScriptTemplate/ios/ReactNativeTypeScriptTemplate/InitialViewController.h [hunk 12]
```diff
@@ -0,0 +1,29 @@
+/*
+ Copyright (c) 2013-present, salesforce.com, inc. All rights reserved.
```
Comment (bhariharan) [inline ReactNativeTypeScriptTemplate/ios/ReactNativeTypeScriptTemplate/InitialViewController.h:2]:
```suggestion
 Copyright (c) 2020-present, salesforce.com, inc. All rights reserved.
```

File: ReactNativeTypeScriptTemplate/ios/ReactNativeTypeScriptTemplate/InitialViewController.m [hunk 13]
```diff
@@ -0,0 +1,50 @@
+/*
+ Copyright (c) 2013-present, salesforce.com, inc. All rights reserved.
```
Comment (bhariharan) [inline ReactNativeTypeScriptTemplate/ios/ReactNativeTypeScriptTemplate/InitialViewController.m:2]:
```suggestion
 Copyright (c) 2020-present, salesforce.com, inc. All rights reserved.
```

File: ReactNativeTypeScriptTemplate/ios/ReactNativeTypeScriptTemplate/main.m [hunk 14]
```diff
@@ -0,0 +1,33 @@
+/*
+ Copyright (c) 2015-present, salesforce.com, inc. All rights reserved.
```
Comment (bhariharan) [inline ReactNativeTypeScriptTemplate/ios/ReactNativeTypeScriptTemplate/main.m:2]:
```suggestion
 Copyright (c) 2020-present, salesforce.com, inc. All rights reserved.
```

File: ReactNativeTypeScriptTemplate/template.js [hunk 15]
```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright (c) 2016-present, salesforce.com, inc.
```
Comment (bhariharan) [inline ReactNativeTypeScriptTemplate/template.js:2]:
```suggestion
 * Copyright (c) 2020-present, salesforce.com, inc.
```

**Review by bhariharan — CHANGES_REQUESTED** [review]
You can directly commit most of my suggestions (pretty cool new `Github` feature) in a single commit. Also, the new template needs to be added to the list in `setversion.sh`. Looks fine, otherwise.

**Comment by wmathurin** [comment]
I copied ReactNativeTemplate so the same typos exist there as well :-(

**Pull request merged**